src/ngx_http_lua_balancer.c

```diff
@@ -266,6 +267,10 @@ ngx_http_lua_balancer_get_peer(ngx_peer_connection_t *pc, void *data)
     bp->more_tries = 0;
     bp->total_tries++;
 
+    lmcf = ngx_http_get_module_main_conf(r, ngx_http_lua_module);
+
+    lmcf->balancer_peer = bp;
```

Comment:
Better add a brief comment explaining why it is safe to store the peer data in the main conf.